platform/autoupdate.services/src/org/netbeans/api/autoupdate/OperationContainer.java

```diff
@@ -352,6 +353,17 @@ public void removeAll() {
         impl.removeAll ();
     }
     
+    /** Specifies location of unpack200 executable. {@code unpack200} has been
+     * removed from JDK 14. As such it is not possible to unpack older NBM
+     * files without providing alternative JDK implementation of this file.
+     *
+     * @param executable path to the executable
+     * @since 1.65
+     */
+    public final void setUnpack200(File executable) {
+        this.impl.setUnpack200(executable);
+    }
```

Comment:
FWIW the pack200 in JDK is not in Java, it is a C/C++ implementation. So seems likely it would be run as a command line tool if we included it.

Reply:
Whether it it is java or not - we can call it via CLI, and most probably that is the best idea, but it looks as if there is a java implementation:

https://github.com/pack200/pack200/blob/master/src/main/java/io/pack200/UnpackerImpl.java#L114-L117
https://github.com/pack200/pack200/blob/master/src/main/java/io/pack200/UnpackerImpl.java#L200

It was just a thought.

Reply:
@matthiasblaesing yes, the mix of GPLv2 and GPLv2-CPE in the JDK is a problem (although it's possible OpenJ9 wouldn't be).  I don't think you can separate that fact from considerations of linking vs aggregation.  Relevance?  Linking in pack200 or nb-javac might be possible given current discussions on GPL+CPE.  Calling the CLI tool as @jlahoda suggests might not allow us to include it.

Reply:
I suggest to take the licensing discussions elsewhere off this PR. It is not intention of this PR to bundle `unpack200` executable. The goal of this PR is to:
* warn user
*  allow the user to provide alternative implementation of the executable
The assumption is that most of the developers find/install older JDK and use its executable.

Reply:
@JaroslavTulach in general agreed. Mentioned because this seems to lock the idea it's an external executable into the API? Whereas something that allowed specifying source and dest files might not - eg. alternative implementations for ModuleUpdate::unpack200 passed in to API?